### ISSUE TITLE
Spreadsheet: Stable circular definitions aren't errors

### ISSUE THREAD

**Issue by water-ghosts** [SerenityOS/serenity#13735]
- In cell A0, type a literal number (e.g. 5).
- In cell A1, type the formula `=A0`.
- Edit cell A0 to use the formula `=A1`.

Despite the circular reference, both cells still have the value 5:

![circular1](https://user-images.githubusercontent.com/56945701/163753094-d327c78c-50b1-454e-bf66-9ffc693e8601.png)
![circular2](https://user-images.githubusercontent.com/56945701/163753098-1c0a7b19-8f31-47ea-b92b-b818c7f5566b.png)

If the circular definition isn't stable (e.g. `A1 = A0 + 1` instead of `A1 = A0`), both cells instead get the value `NaN`:

![unstable](https://user-images.githubusercontent.com/56945701/163753709-fae36778-4239-40c8-9faa-84fa39d3cffa.png)

In contrast, Google Sheets treats all circular definitions as errors:

![circular3](https://user-images.githubusercontent.com/56945701/163753881-ae78ccef-3c9e-4c97-9b79-a70bb5070918.png)

This is arguably a feature, not a bug. If this is intended behavior, feel free to close this issue!

**Comment by Diogenes31** [comment]
I agree that it's a bug, I think that could create some very easy to miss errors in a spreadsheet that could cause much headache.

**Comment by kleinesfilmroellchen** [comment]
This does indeed look like a bug, though we want @alimpfard 's opinion here. We should display a custom error instead imho.
